src/styles/app-text-input/AppTextInput.scss

```diff
@@ -0,0 +1,84 @@
+$icon-size: 20px;
+
+.app-input {
+  display: inline-flex;
+  flex-direction: column;
+  gap: 4px;
+
+  &__control-container {
+    position: relative;
+    align-self: flex-start;
+    min-width: var(--min-width); // from react props
+
+    &:has(.app-input__start-adornment) {
+      input {
+        padding-left: 48px;
+      }
+    }
+
+    &:has(.app-input__end-adornment) {
+      input {
+        padding-right: 48px;
+      }
+    }
+  }
+
+  &__adornment {
+    position: absolute;
+    top: 50%;
+    transform: translateY(-50%);
+    width: $icon-size;
+    height: $icon-size;
+  }
+
+  &__start-adornment {
+    left: 16px;
+  }
+
+  &__end-adornment {
+    right: 16px;
+  }
+
+  &--full-width {
+    display: flex;
+
+    .app-input__control-container {
+      align-self: stretch;
+    }
+  }
+
+  &__filled {
+    border: 1px solid black;
+  }
+
+  &__underlined {
```

Comment:
```suggestion
    &__underlined {
    transition: all 0.3s ease;
    border: none;
    border-bottom: 1px solid black;

    &:focus-visible {
      outline: none;
      transform: scaleX(1.05);
    }
  }
```

![image](https://github.com/user-attachments/assets/48188b3d-5364-469a-b0a9-164fad298f28)

The underlined variant doesn't apply any styles on focusing. I think, we can ask out designed to create styles for focus (or I suggest to use my variant of animation)

Reply:
We need to ask designer about that. I left a todo. I guess after deploying storybook we can do more detailed testing and detect some issues